cli/tests/it/cmd.rs

```diff
@@ -30,6 +30,13 @@ forgetest!(can_clean_non_existing, |prj: TestProject, mut cmd: TestCommand| {
     prj.assert_cleaned();
 });
 
+// checks that `cache clean` can be invoked and cleans the foundry cache
+forgetest!(can_cache_clean, |prj: TestProject, mut cmd: TestCommand| {
+    cmd.args(["cache", "clean"]);
+    cmd.assert_empty_stdout();
+    prj.assert_foundry_cache_cleaned();
+});
+
```

Comment:
great!
could we also mock some empty block files here?
just creating empty files for a chain and test with chain and block argument as well?

Reply:
added tests for `cache clean <chain>` and `cache clean <chain> --block <blocks>`. in each test I'm setting up directories at the start of the test so our assertions that the paths don't exist won't pass by default.

any preference or pointers on how to best isolate these tests? rn they operate on `~` and can fail if run in parallel. it looks like there are crates for mocking or working with a [fake file system](https://github.com/iredelmeier/filesystem-rs) but didn't see any of them being used by the project currently